packages/next/src/lib/find-root.ts

```diff
@@ -23,21 +23,24 @@ export function findRootDir(cwd: string) {
 
   const lockFiles = [lockFile]
   while (true) {
-    const nextDir = dirname(dirname(lockFiles[lockFiles.length - 1]))
-    const newLockFile = findRootLockFile(nextDir)
+    const lastLockFile = lockFiles[lockFiles.length - 1]
+    const currentDir = dirname(lastLockFile)
+    const parentDir = dirname(currentDir)
 
-    if (newLockFile) {
-      lockFiles.push(newLockFile)
-    } else {
-      break
-    }
+    if (parentDir === currentDir) break
+
+    const newLockFile = findRootLockFile(parentDir)
+
+    if (!newLockFile || newLockFile === lastLockFile) break
```

Comment:
is `newLockFile === lastLockFile` ever possible? since parentDir is definitely a parent dir of last lock file and findRootLockFile only looks 'up'?

Reply:
MMm that's right - I'll update